### PR TITLE
[docs] README update

### PR DESCRIPTION
**Pull request by stevhliu**
This PR cleans up the `README.md` as a whole:

➕ adds a tip to promote and watch the PEFT Hub organization more for updates on the latest supported PEFT methods and notebooks
➕ expands quickstart a bit with installation, training, and inference code snippets
➕ creates two new sections, Why you should use PEFT (this focuses on the benefits and use cases) and PEFT integrations (this focuses on how PEFT fits into the 🤗  ecosystem)
➕ adds the Space we have in the docs for discovering supporting models (more compact than the current model matrix)
➖ removes specific code snippets of using Accelerate and DreamBooth training, caveats, injecting adapter types, and mixing different adapter types (all of this info can already be found in the docs)

### CODE REVIEW

**Review by younesbelkada — COMMENTED** [review]
Thanks so much @stevhliu for this! 
I left few suggestions to reflect the README with recent resources such as DPO, etc. Please let me know what do you think

File: README.md [hunk 1]
```diff
@@ -19,25 +19,36 @@ limitations under the License.
...
+3. P-Tuning: [GPT Understands, Too](https://hf.co/papers/2103.10385)
+4. Prompt Tuning: [The Power of Scale for Parameter-Efficient Prompt Tuning](https://hf.co/papers/2104.08691)
+5. AdaLoRA: [Adaptive Budget Allocation for Parameter-Efficient Fine-Tuning](https://hf.co/papers/2303.10512)  
+6. $(IA)^3$: [Few-Shot Parameter-Efficient Fine-Tuning is Better and Cheaper than In-Context Learning](https://hf.co/papers/2205.05638)
+7. MultiTask Prompt Tuning: [Multitask Prompt Tuning Enables Parameter-Efficient Transfer Learning](https://hf.co/papers/2303.02861)
+8. LoHa: [FedPara: Low-Rank Hadamard Product for Communication-Efficient Federated Learning](https://hf.co/papers/2108.06098)
+9. LoKr: [KronA: Parameter Efficient Tuning with Kronecker Adapter](https://hf.co/papers/2212.10650) based on [Navigating Text-To-Image Customization:From LyCORIS Fine-Tuning to Model Evaluation](https://hf.co/papers/2309.14859) implementation
+10. LoftQ: [LoftQ: LoRA-Fine-Tuning-aware Quantization for Large Language Models](https://hf.co/papers/2310.08659)
+11. OFT: [Controlling Text-to-Image Diffusion by Orthogonal Finetuning](https://hf.co/papers/2306.07280)
```
Comment (pacman100) [inline README.md:41]:
```suggestion
11. OFT: [Controlling Text-to-Image Diffusion by Orthogonal Finetuning](https://hf.co/papers/2306.07280)
12. Polytropon: [Combining Modular Skills in Multitask Learning](https://arxiv.org/abs/2202.13914) and Multi-Head Routing: [Multi-Head Adapter Routing for Cross-Task Generalization](https://arxiv.org/abs/2211.03831)
13. AdaptationPrompt/LlamaAdapter: [LLaMA-Adapter: Efficient Fine-tuning of Language Models with Zero-init Attention](https://arxiv.org/abs/2303.16199)
```

**Review by pacman100 — COMMENTED** [review]
This is great 🧹✨🤗! Thank you @stevhliu for refactoring the README which has been pending for some time now. I like the changes and overall everything LGTM. I also like the suggestions from Younes.

I've given a suggestion.

File: README.md [hunk 2]
```diff
@@ -52,395 +63,108 @@ peft_config = LoraConfig(
...
 
-model = DummyModel()
-model = inject_adapter_in_model(lora_config, model)
+> [!TIP]
+> Take a look at the [examples/lora_dreambooth/train_dreambooth.py](examples/lora_dreambooth/train_dreambooth.py) training script to try training your own Stable Diffusion model with LoRA, and play around with the [smangrul/peft-lora-sd-dreambooth](https://huggingface.co/spaces/smangrul/peft-lora-sd-dreambooth) Space which is running on a T4 instance.
 
-dummy_inputs = torch.LongTensor([[0, 1, 2, 3, 4, 5, 6, 7]])
-dummy_outputs = model(dummy_inputs)
-```
+### Accelerate
```
[the remark this answers is not in the document]
Comment (stevhliu) [reply README.md:139]: Added it to the Tip for more visibility!

**Comment by stevhliu** [comment]
Good idea, I think it's better to link users to existing resources instead of maintaining another list :)

**Pull request merged**